programs/wen_royalty_distribution/src/instructions/update.rs

```diff
@@ -42,33 +41,72 @@ pub struct UpdateDistribution<'info> {
         mint::token_program = anchor_spl::token_interface::spl_token_2022::id(),
     )]
     pub mint: Box<InterfaceAccount<'info, Mint>>,
-    #[account(mut)]
+    /// CHECK: can be Pubkey::default() or mint address
+    #[account()]
+    pub payment_mint: UncheckedAccount<'info>,
+    #[account(
+        mut,
+        has_one = payment_mint,
+        seeds = [distribution_account.group_mint.as_ref(), payment_mint.key().as_ref()],
+        bump
+    )]
```

Comment:
If seed are checked in TE and it has access to the bump on the account then should not redo the check and have to find the bump on each instruciton.